azurelinuxagent/daemon/resourcedisk/default.py

```diff
@@ -174,6 +174,24 @@ def mount_resource_disk(self, mount_point, fs):
                     fs)
         return mount_point
 
+    def change_partition_type(self, suppress_message, option_str):
+        """
+            use sfdisk to change partition type.
+            As --part-type is supported in sfdisk 2.26+,
+            use -c for lower versions
+        """
+
+        command_to_use = '--part-type'
+        # sfdisk version string is in the following formats:
+        # 1. "sfdisk from util-linux 2.27.1"
+        # 2. "sfdisk (util-linux-ng 2.17.2)"
+        err_code, version_str = shellutil.run_get_output("sfdisk -v")
+        if err_code == 0 and version_str.split()[-1].strip('(), ') < '2.26':
```

Comment:
I think if you are going to rely on this type of check you would need to ensure the version string matches first, as it might change in the future, so it might be better to do this kind of check with a regex.